csm/core/agent/api.py

```diff
@@ -249,31 +249,63 @@ def getMatchingEndpoint(endpoint_map, path):
         else:
             Log.debug(f"Feature endpoint is not found for {request.path}")
 
+    @classmethod
+    def _extract_bearer(cls, headers) -> str:
+        """
+        Extract the bearer token from HTTP headers.
+
+        :param headers: HTTP headers.
+        :returns: bearer token.
+        """
+
+        hdr = headers.get(CsmAuth.HDR)
+        if not hdr:
+            raise CsmNotFoundError(f'No {CsmAuth.HDR} header')
+        auth_pair = hdr.split(' ')
+        if len(auth_pair) != 2:
+            raise CsmNotFoundError(f'The header is incorrect. Expected "{CsmAuth.HDR} session_id"')
+        auth_type, session_id = auth_pair
+        if auth_type != CsmAuth.TYPE:
+            raise CsmNotFoundError(f'Invalid auth type {auth_type}')
+        return session_id
+
+    @classmethod
+    async def _validate_bearer(cls, login_service, session_id):
+        """
+        Validate the bearer token.
+
+        Search for the session with ID equal to the token value and return it.
+        Throw a Permission denied exception otherwise.
+        :param login_service: login service.
+        :param session_id: bearer token's value.
+        :returns: session object.
+        """
+
+        try:
+            session = await login_service.auth_session(session_id)
+            Log.info(f'Username: {session.credentials.user_id}')
+        except CsmError as e:
+            raise CsmNotFoundError(e.error())
+        if not session:
+            raise CsmNotFoundError('Invalid auth token')
```

Comment:
Maybe the
```
Log.info(f'Username: {session.credentials.user_id}')
```
In the `try` block should be placed after this line.

Reply:
Done.

Reply:
This change has not been committed, I think, I can't see it in the changeset and it is not marked as "outdated" as well.

Reply:
Oh, sorry. Amended the last commit.